File: ruby/stage2.rb
require 'http'
require './api'

module Stage2
	def self.runstage2(token)
		data = API.communicate('haystack', {"token" => token})

		haystack = data['haystack']
		needle = data['needle']

		index = haystack.index(needle)
		API.communicate('validateneedle', {'token' => token, 'needle' => index})
	end
end

File: python/stage2.py
import api
import json

def runstage2(token):
	# get data
	jsonData = api.communicate('haystack', {'token':token})

	# extract data
	haystack = jsonData['haystack']
	needle = jsonData['needle']

	# get index, according to instructions it is always there, but lets
	# just add some basic error handling
	try:
		index = haystack.index(needle)
	except:
		print "The needle is not in the haystack"

	# validate result
	return api.communicate('validateneedle', {'token':token, 'needle':index})


